apps/webservice/src/app/[workspaceSlug]/(targets)/targets/TargetPageContent.tsx

```diff
@@ -27,54 +32,62 @@ export const TargetPageContent: React.FC<{
   view: schema.TargetView | null;
 }> = ({ workspace, view }) => {
   const { filter, setFilter, setView } = useTargetFilter();
+  const workspaceId = workspace.id;
+  const targetsAll = api.target.byWorkspaceId.list.useQuery({ workspaceId });
+  const targets = api.target.byWorkspaceId.list.useQuery(
+    { workspaceId, filter },
+    { placeholderData: (prev) => prev },
+  );
 
-  const targetsAll = api.target.byWorkspaceId.list.useQuery({
-    workspaceId: workspace.id,
-  });
-
-  const targets = api.target.byWorkspaceId.list.useQuery({
-    workspaceId: workspace.id,
-    filter,
-  });
+  const onFilterChange = (condition: TargetCondition | undefined) => {
+    const cond = condition ?? defaultCondition;
+    if (isEmptyCondition(cond)) setFilter(undefined);
+    if (!isEmptyCondition(cond)) setFilter(cond);
+  };
 
   const { targetId, setTargetId } = useTargetDrawer();
 
   if (targetsAll.isSuccess && targetsAll.data.total === 0)
     return <TargetGettingStarted />;
   return (
     <div className="h-full text-sm">
-      <div className="flex items-center justify-between border-b border-neutral-800 p-1 px-2">
-        <TargetConditionDialog condition={filter} onChange={setFilter}>
-          <div className="flex items-center gap-2">
-            {view == null && (
-              <Button
-                variant="ghost"
-                size="icon"
-                className="flex h-7 w-7 flex-shrink-0 items-center gap-1 text-xs"
-              >
-                <IconFilter className="h-4 w-4" />
-              </Button>
-            )}
+      <div className="flex h-[41px] items-center justify-between border-b border-neutral-800 p-1 px-2">
```

Comment:
explicitly set height to 41px. before, because the count badge didnt exist during the first load, it was causing a small but noticeable shift in height.